src/cpu.c

```diff
@@ -814,8 +814,7 @@ static int cpu_read (void)
 
 	if (pnumcpu != numcpu || perfcpu == NULL)
 	{
-		if (perfcpu != NULL)
-			free(perfcpu);
+		free(perfcpu);
```

Comment:
> I doubt avoiding the function call has a measurable performance impact in practice.

+1. The important thing about `sfree()` is the `(ptr) = NULL;` bit, i.e. that pointers are guaranteed to be invalid afterwards. If you feel strongly about the `NULL` check, feel free to send a PR.